File: apps/watchman/lib/watchman/forge/queries.ex
defmodule Watchman.Forge.Queries do
  @dashboard_fragment """
    fragment DashboardFragment on Dashboard {
      name
      uid
    }
  """

  @repository_fragment """
    fragment RepositoryFragment on Repository {
      id
      name
      icon
      description
      dashboards {
        ...DashboardFragment
      }
    }
    #{@dashboard_fragment}
  """

  @installation_fragment """
    fragment InstallationFragment on Installation {
      id
      repository {
        ...RepositoryFragment
      }
    }
    #{@repository_fragment}
  """

  @installation_query """
    query Installations($first: Int!, $cursor: String) {
      installations(first: $first, after: $cursor) {
        pageInfo {
          endCursor
          hasNextPage
        }
        edges {
          node {
            ...InstallationFragment
          }
        }
      }
    }
    #{@installation_fragment}
  """

  def installation_query(), do: @installation_query
end

File: www/src/components/cluster/queries.ts
import { gql } from '@apollo/client'

import { PageInfo } from '../../models/misc'

export const REPOSITORIES_Q = gql`
  query Repositories($cursor: String, $installed: Boolean) {
    repositories(after: $cursor, first: 200, installed: $installed) {
      pageInfo {
        ...PageInfo
      }
      edges {
        node {
          id
          name
          icon
          darkIcon
          installation {
            pingedAt
          }
        }
      }
    }
  }
  ${PageInfo}
`
